internal/operators/validate_schema.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2022 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build !tinygo && !coraza.disabled_operators.validateSchema
+// +build !tinygo,!coraza.disabled_operators.validateSchema
+
+package operators
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/corazawaf/coraza/v3/experimental/plugins/plugintypes"
+	"github.com/kaptinlin/jsonschema"
+)
+
+type validateSchema struct {
+	schemaType string
+	schemaPath string
+	schemaData []byte
+	jsonSchema *jsonschema.Schema
+	initOnce   sync.Once
+	initError  error
+}
+
+var _ plugintypes.Operator = (*validateSchema)(nil)
+
+// NewValidateSchema creates a new validateSchema operator
+func NewValidateSchema(options plugintypes.OperatorOptions) (plugintypes.Operator, error) {
+	data := options.Arguments
+	if data == "" {
+		return nil, errors.New("schema file path is required")
+	}
+
+	var schemaData []byte
+	var err error
+
+	// Check if it's a file path and read schema data
+	if options.Root != nil {
+		// Handle file from provided filesystem root
+		schemaData, err = fs.ReadFile(options.Root, data)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read schema file from root FS: %v", err)
+		}
+	} else {
+		// Direct file access fallback
+		schemaData, err = os.ReadFile(data)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read schema file: %v", err)
+		}
+	}
+
+	// Determine schema type from file extension
+	ext := strings.ToLower(filepath.Ext(data))
+	if ext != ".json" {
+		return nil, fmt.Errorf("unsupported schema type: %s, must be .json", ext)
+	}
+
+	// Preliminarily validate that the schema is valid JSON
+	var jsonSchema interface{}
+	if err := json.Unmarshal(schemaData, &jsonSchema); err != nil {
+		return nil, fmt.Errorf("invalid JSON schema: %v", err)
+	}
+
+	operator := &validateSchema{
+		schemaType: "json",
+		schemaPath: data,
+		schemaData: schemaData,
+	}
+
+	return operator, nil
+}
+
+// initValidators lazily initializes the validators to avoid doing expensive operations during initialization
+func (o *validateSchema) initValidators() error {
+	var err error
+	o.initOnce.Do(func() {
+		// Initialize JSON Schema validator
+		compiler := jsonschema.NewCompiler()
+
+		// Parse the schema
+		var schema *jsonschema.Schema
+		schema, err = compiler.Compile(o.schemaData)
+		if err != nil {
+			o.initError = fmt.Errorf("failed to compile JSON schema: %v", err)
+			return
+		}
+		o.jsonSchema = schema
+	})
+	return o.initError
+}
+
+func (o *validateSchema) Evaluate(tx plugintypes.TransactionState, data string) bool {
+	// Lazy initialize the validators
```

Comment:
I wonder if we should be doing this lazily. Can't we do in bootstrap instead of adding a lock and attempt to initialize on every request?

Reply:
Done.